narwhals/_expression_parsing.py

```diff
@@ -384,68 +389,85 @@ def __repr__(self) -> str:  # pragma: no cover
             f"  preserves_length: {self.preserves_length},\n"
             f"  is_scalar_like: {self.is_scalar_like},\n"
             f"  is_literal: {self.is_literal},\n"
-            f"  nodes: {self.nodes},\n"
+            f"  nodes: {tuple(self.iter_nodes())},\n"
             ")"
         )
 
+    def iter_nodes(self) -> Iterator[ExprNode]:
```

Comment:
`iter_nodes` is called only in `__repr__`, which itself is not covered to begin with

```suggestion
    def iter_nodes(self) -> Iterator[ExprNode]:  # pragma: no cover
```